### PR TITLE
mame & rom-tools 0.209

### PR DESCRIPTION
**Pull request by mbcoguno**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
* Use Homebrew's asio, gdlm, pugixml and rapidjson instead of mame bundled in build stage.
* Remove sphinx-doc's additional extension in docs config to prevent error when loading missing extension.
* Remove unused runtime dependent.
* Add testkeys binary.
* ~~Use Homebrew's compiler.~~

### CODE REVIEW

**Comment by mbcoguno** [comment]
I don't know why it can't build on sierra.

**Comment by zbeekman** [comment]
@BrewTestBot test this please

File: Formula/mame.rb [hunk 1]
```diff
@@ -13,39 +13,49 @@ class Mame < Formula
...
                    "USE_SYSTEM_LIB_FLAC=1",
+                   "USE_SYSTEM_LIB_GLM=1",
+                   "USE_SYSTEM_LIB_JPEG=1",
                    "USE_SYSTEM_LIB_LUA=1",
-                   "USE_SYSTEM_LIB_SQLITE3=1",
                    "USE_SYSTEM_LIB_PORTMIDI=1",
                    "USE_SYSTEM_LIB_PORTAUDIO=1",
+                   "USE_SYSTEM_LIB_PUGIXML=1",
+                   "USE_SYSTEM_LIB_RAPIDJSON=1",
+                   "USE_SYSTEM_LIB_SQLITE3=1",
```
Comment (zbeekman) [inline Formula/mame.rb]: Thanks for alphabetizing these.

**Comment by zbeekman** [comment]
For mame:
```
Linking mame64...
Undefined symbols for architecture x86_64:
  "pstring_t<putf8_traits>::operator=(pstring_t<putf8_traits>&&)", referenced from:
      netlist_mame_stream_input_device::custom_netlist_additions(netlist::netlist_state_t&) in liboptional.a(netlist.o)
      std::__1::vector<std::__1::pair<pstring_t<putf8_traits>, plib::owned_ptr<netlist::core_device_t, plib::arena_deleter<plib::mempool, netlist::core_device_t> > >, std::__1::allocator<std::__1::pair<pstring_t<putf8_traits>, plib::owned_ptr<netlist::core_device_t, plib::arena_deleter<plib::mempool, netlist::core_device_t> > > > >::insert(std::__1::__wrap_iter<std::__1::pair<pstring_t<putf8_traits>, plib::owned_ptr<netlist::core_device_t, plib::arena_deleter<plib::mempool, netlist::core_device_t> > > const*>, std::__1::pair<pstring_t<putf8_traits>, plib::owned_ptr<netlist::core_device_t, plib::arena_deleter<plib::mempool, netlist::core_device_t> > >&&) in liboptional.a(netlist.o)
```

For rom-tools:
```
Linking nltool...
Undefined symbols for architecture x86_64:
  "pstring_t<putf8_traits>::operator=(pstring_t<putf8_traits>&&)", referenced from:
      void std::__1::__sort<tool_app_t::create_docheader()::$_0&, pstring_t<putf8_traits>*>(pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, tool_app_t::create_docheader()::$_0&) in nltool.o
      unsigned int std::__1::__sort3<tool_app_t::create_docheader()::$_0&, pstring_t<putf8_traits>*>(pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, tool_app_t::create_docheader()::$_0&) in nltool.o
      unsigned int std::__1::__sort4<tool_app_t::create_docheader()::$_0&, pstring_t<putf8_traits>*>(pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, tool_app_t::create_docheader()::$_0&) in nltool.o
      unsigned int std::__1::__sort5<tool_app_t::create_docheader()::$_0&, pstring_t<putf8_traits>*>(pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, tool_app_t::create_docheader()::$_0&) in nltool.o
      bool std::__1::__insertion_sort_incomplete<tool_app_t::create_docheader()::$_0&, pstring_t<putf8_traits>*>(pstring_t<putf8_traits>*, pstring_t<putf8_traits>*, tool_app_t::create_docheader()::$_0&) in nltool.o
      std::__1::vector<pstring_t<putf8_traits>, std::__1::allocator<pstring_t<putf8_traits> > >::erase(std::__1::__wrap_iter<pstring_t<putf8_traits> const*>, std::__1::__wrap_iter<pstring_t<putf8_traits> const*>) in nltool.o
      std::__1::__wrap_iter<pstring_t<putf8_traits>*> std::__1::remove<std::__1::__wrap_iter<pstring_t<putf8_traits>*>, pstring_t<putf8_traits> >(std::__1::__wrap_iter<pstring_t<putf8_traits>*>, std::__1::__wrap_iter<pstring_t<putf8_traits>*>, pstring_t<putf8_traits> const&) in nltool.o
      ...
ld: symbol(s) not found for architecture x86_64
```

**Comment by zbeekman** [comment]
A quick and dirty work around would be to make them both depend on high_sierra and later... not ideal.

```ruby
depends_on :macos => :high_sierra
```

**Comment by mbcoguno** [comment]
Sorry I'm late.
Yesterday I test build on Sierra VM with `--cc=llvm_clang` and add `OVERRIDE_CC` in make flags and it works. Maybe it fails with Sierra's Clang.
Tomorrow I will update commits to using Homebrew's compiler, but it will take about extra one hour to build.

**Comment by zbeekman** [comment]
> Yesterday I test build on Sierra VM with `--cc=llvm_clang` and add `OVERRIDE_CC` in make flags and it works. Maybe it fails with Sierra's Clang.
> Tomorrow I will update commits to using Homebrew's compiler, but it will take about extra one hour to build.

At this point, I'd rather set 
```
depends_on :macos => :high_sierra
```
get rid of the
```
fails_with :clang do
```
and be done with it.

**Review by zbeekman — APPROVED** [review]
LGTM

**Comment by zbeekman** [comment]
Thanks for your contribution to Homebrew! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**